fitbenchmarking/cli/main.py

```diff
@@ -81,12 +81,72 @@ def get_parser():
+                        nargs='+',
+                        type = str,
+                        default = [],
+                        help = "Select what type of algorithm is used within a specific software.")
+    parser.add_argument('-s', '--software',
+                        metavar = 'SOFTWARE',
+                        nargs='+',
+                        type = str,
+                        default = [],
+                        help = "Select the fitting software to benchmark.")    
+    parser.add_argument('-j', '--jac_method',
+                        metavar = 'JAC_METHOD',
+                        nargs='+',
+                        type = str,
+                        default = [],
+                        help = "Set the Jacobian to be used.")
+    parser.add_argument('-c', '--cost_func_type',
+                        metavar = 'COST_FUNC_TYPE',
+                        nargs='+',
+                        type = str,
+                        default = [],
+                        help = "Set the cost functions to be used for the given data.")
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--do_make_plots', action = 'store_true')
+    group.add_argument('--dont_make_plots', action = 'store_true')
+
+    parser.add_argument('-m', '--comparison_mode',
+                        metavar = 'COMPARISON_MODE',
+                        default = '',
+                        help = "Select the mode for displaying values in the resulting table.")               
+    parser.add_argument('-t', '--table_type',
+                        metavar = 'TABLE_TYPE',
+                        nargs='+',
+                        type = str,
+                        default = [],
+                        help = "Select the type of table to be produced in FitBenchmarking.")                
+    parser.add_argument('-f', '--logging_file_name',
+                        metavar = 'LOGGING_FILE_NAME',
+                        default = '',
+                        help = "Specify the file path to write the logs to.") 
+
+    group2 = parser.add_mutually_exclusive_group()
+    group2.add_argument('--do_append', action = 'store_true')
+    group2.add_argument('--dont_append', action = 'store_true')
+
+    parser.add_argument('-l', '--level',
+                        metavar = 'LEVEL',
+                        default = '',
+                        help = "Specify the minimum level of logging to display on console during runtime.")  
+    parser.add_argument('-e', '--external_output',
+                        metavar = 'EXTERNAL_OUTPUT',
+                        default = '',
+                        help = "Select the amount of information displayed from third-parties.")                       
     return parser
 
 
-def _find_options_file(options_file: str, results_directory: str) -> Options:
+def _find_options_file(options_file: str, results_directory: str, additional_options : dict) -> Options:
```

Comment:
The in-code documentation (text in the triple quotes) for this will need updating to explain what additional_options is.